tex/generic/pgf/frontendlayer/tikz/libraries/tikzlibrarymindmap.code.tex

```diff
@@ -17,72 +17,63 @@
-    \pgfpatharc{\pgfdecorationsegmentangle}{-\pgfdecorationsegmentangle}{\pgfkeysvalueof{/pgf/decoration/start radius}}
-    \pgfutil@tempcnta=-\pgfdecorationsegmentangle\relax
-    \advance\pgfutil@tempcnta by90\relax
     \pgfmathsetlength\pgfutil@tempdima{\pgfkeysvalueof{/pgf/decoration/start radius}}
     \pgfmathsetlength\pgfutil@tempdimb{\pgfdecorationsegmentamplitude}
-    \pgfpathcurveto
-    {\pgfpointadd
-      {\pgfpointpolar{-\pgfdecorationsegmentangle}{\pgfkeysvalueof{/pgf/decoration/start radius}}}
-      {\pgfpointpolar{\the\pgfutil@tempcnta}{.25\pgfutil@tempdima}}}
-    {\pgfqpoint{1.25\pgfutil@tempdima}{-.5\pgfutil@tempdimb}}
-    {\pgfqpoint{1.5\pgfutil@tempdima}{-.5\pgfutil@tempdimb}}
-    \pgfpathlineto{\pgfpoint{1.5\pgfutil@tempdima}{.5\pgfutil@tempdimb}}
+    \pgfmathsetlength\pgf@lib@mm@flarelength{\pgfkeysvalueof{/pgf/decoration/start flare length}}
+    \pgfpathmoveto{\pgfpoint{\pgfutil@tempdima+1.0\pgf@lib@mm@flarelength}{.5\pgfutil@tempdimb}}
     \pgfutil@tempcnta=\pgfdecorationsegmentangle\relax
     \advance\pgfutil@tempcnta by-90\relax
     \pgfpathcurveto
-    {\pgfpoint{1.25\pgfutil@tempdima}{.5\pgfutil@tempdimb}}
+    {\pgfpoint{\pgfutil@tempdima+0.5\pgf@lib@mm@flarelength}{.5\pgfutil@tempdimb}}
     {\pgfpointadd
       {\pgfpointpolar{\pgfdecorationsegmentangle}{\pgfkeysvalueof{/pgf/decoration/start radius}}}
       {\pgfpointpolar{\the\pgfutil@tempcnta}{.25\pgfutil@tempdima}}}
     {\pgfpointpolar{\pgfdecorationsegmentangle}{\pgfkeysvalueof{/pgf/decoration/start radius}}}
-    \pgfpathclose
+    \pgfpathlineto{\pgfpointpolar{\pgfdecorationsegmentangle}{\pgfkeysvalueof{/pgf/decoration/start radius}}}
+    \pgfpatharc{\pgfdecorationsegmentangle}{-\pgfdecorationsegmentangle}{\pgfkeysvalueof{/pgf/decoration/start radius}}
+    \pgfutil@tempcnta=-\pgfdecorationsegmentangle\relax
+    \advance\pgfutil@tempcnta by90\relax
+    \pgfpathcurveto
+    {\pgfpointadd
+      {\pgfpointpolar{-\pgfdecorationsegmentangle}{\pgfkeysvalueof{/pgf/decoration/start radius}}}
+      {\pgfpointpolar{\the\pgfutil@tempcnta}{.25\pgfutil@tempdima}}}
+    {\pgfpoint{\pgfutil@tempdima+0.5\pgf@lib@mm@flarelength}{-.5\pgfutil@tempdimb}}
+    {\pgfpoint{\pgfutil@tempdima+1.0\pgf@lib@mm@flarelength}{-.5\pgfutil@tempdimb}}
     }
-  }%
-  \state{bar}[width=0pt,next state=end]
-  {
-    \pgfmathsetlength\pgfutil@tempdima{\pgfkeysvalueof{/pgf/decoration/start radius}}%
-    \pgfmathsetlength\pgfutil@tempdimb{\pgfkeysvalueof{/pgf/decoration/end radius}}%
-    \pgfmathsetlength\pgf@xc{\pgfdecorationsegmentamplitude}%
-    \pgfpathrectangle
-    {\pgfqpoint{.5\pgfutil@tempdima}{-.5\pgf@xc}}
-    {\pgfpoint{\pgfdecoratedremainingdistance+-.5\pgfutil@tempdimb+-.5\pgfutil@tempdima}{\pgf@xc}}
-  }%
-  \state{end}[width=0pt,next state=final]
-  {
     {
     \pgftransformxshift{\pgfdecoratedremainingdistance}%
     \pgftransformxscale{-1}%
+    \pgftransformyscale{-1}%
```

Comment:
This will reset the xscale.  Did you mean this?
```suggestion
    \pgftransformscale{-1}%
```

Reply:
One line is better than two, so, good.  But FWIW my experiments don't match your expectation.  If the x scale were reset, then I would expect commenting out line 52 would do nothing; but that is not the case.